docs/zh/reference/ip_tips.md

```diff
@@ -1,39 +1,123 @@
+
+首先，让我们定义下，什么是外部？
+
+- 物理机：一台主机访问另一台主机，就是外部。同一主机上，使用127.0.0.1也可正常通讯；外部则必须使用“被访问主机“的公网IP。
+- 容器：同一主机的容器外，都是外部，包括同一主机的另一个容器、同一主机的物理环境，另外的主机。
+
+其次，让我们明确下，OpenMLDB有哪几种分布形式？
+- onebox，所有OpenMLDB server都在一个环境下，同一物理机或一个容器内。例如，我们的[快速上手](../quickstart/openmldb_quickstart.md)，就是将所有进程都放在一个容器内。
+- 分布式，正式生产环境中常用分布式，server在不同物理机上，它们自然是需要绑定公网IP。
+
+由于容器的网络限制，onebox型的OpenMLDB常出现，IP配置错误等问题。相反，分布式由于一定要绑定公网IP，反而没有太多疑问。
+
+下面我们将介绍**onebox型OpenMLDB**如何修改配置实现**外部访问**。
 ```{attention}
-单机版中，不只是需要改endpoint，nameserver的配置中还有tablet ip `--tablet=`，此处也需要修改。
+单机版中，不只是需要改endpoint，nameserver的配置中的tablet ip `--tablet=`也需要修改。
 ```
 
-## Docker IP
+## Onebox型OpenMLDB外部访问
+
+OpenMLDB有多种访问方式，包括HTTP，多种SDK，以及命令行CLI。
+
+### Http
+
+如果你只需要用restful http接口，那么，只需要考虑apiserver的ip是否可访问。（onebox型OpenMLDB的apiserver与其他server在同一环境下，它可以自由访问其他server）。
+
+可以通过
+```
+curl http://<ip:port>/dbs/foo -X POST -d'{"mode":"online", "sql":"show components"}'
+```
+可以确认apiserver是否正常工作。这里的nameserver、tablet server等ip即使是127.0.0.1，也不会有问题，因为apiserver可以通过127.0.0.1访问到这些server。
+
+#### 物理机onebox apiserver
+
+跨主机访问物理机上的onebox，只需要让apiserver的endpoint（绑定ip）改为公网ip。
+
 
-希望从容器的外部（无论是同一主机还是跨主机）访问容器内，请先
-更改endpoint`127.0.0.1`为 `0.0.0.0`（单机版中`tablet`配置项也需要更改），以避免不必要的麻烦。
+#### 容器onebox apiserver
 
-### 容器外部访问（同一主机）
-在同一主机中，想要从**容器的外部**（物理机或者是其他容器）访问**容器内**启动的OpenMLDB服务端，可以直接使用bridge的方式连接，也可以暴露端口，还可以直接使用host网络模式。
+如果是本机访问容器onebox中的apiserver，可以**任选一种**下面的方式：
+ - 可以通过bridge的方式，只需让apiserver的endpoint改为`0.0.0.0`（也就是绑定容器ip），然后http使用容器ip即可。
+ ```{note}
+ bridge IP通过`docker network inspect bridge`来查看，通过容器ID或Name找到IP。
 
-```{caution}
-Docker Desktop for Mac无法支持从物理机访问容器（以下任何模式都不能），参考[i-cannot-ping-my-containers](https://docs.docker.com/desktop/mac/networking/#i-cannot-ping-my-containers)。
+ Docker Desktop for Mac无法支持从物理机访问容器（以下任何模式都不能），参考[i-cannot-ping-my-containers](https://docs.docker.com/desktop/mac/networking/#i-cannot-ping-my-containers)。
 
 但macOS中，可以从容器内访问其他容器。
+ ```
+ - 暴露端口，也需要修改apiserver的endpoint改为`0.0.0.0`。这样可以使用127.0.0.1或是公网ip访问到apiserver。
+    单机版：
+    ```
+    docker run -p 8080:8080 -it 4pdosc/openmldb:0.6.0 bash
+    ```
+    集群版：
+    ```
+    docker run -p 9080:9080 -it 4pdosc/openmldb:0.6.0 bash
+    ```
+ - 使用host网络，可以不用修改endpoint配置。缺点是容易引起端口冲突。
+    ```
+    docker run --network host -it 4pdosc/openmldb:0.6.0 bash
+    ```
+
+如果是跨主机访问容器onebox中的apiserver，可以**任选一种**下面的方式：
+ - 暴露端口，并修改apiserver的endpoint改为`0.0.0.0`。docker启动详情见上。
+ - 使用host网络，并修改apiserver的endpoint改为`0.0.0.0`或是公网IP。docker启动详情见上。
+
+只需要让apiserver的endpoint（绑定ip）改为公网ip，使它可访问。apiserver与集群内server的交互都在同一台server上，或同一容器内，并不需要更改。
+还是跨主机
+
+### CLI/SDK
+
+如果你需要在外部使用CLI/SDK，情况比只连接apiserver要复杂，需要保证CLI/SDK能访问到tablet server和taskmanager server。
+```{seealso}
+由于server间内部通信是使用`endpoint`绑定的ip通信，而CLI/SDK也是直接获取同样的ip，直连tablet server或taskmanager，因此，nameserver和tablet/taskmanager通信正常，CLI/SDK却有可能因为跨主机或容器，无法正常连接到tablet/taskmanager。
 ```
 
-#### bridge连接
-bridge连接不需要更改docker run命令，只需要查询一下bridge ip。
+你可以通过这样一个简单的sql脚本来测试确认连接是否正常。
```

Comment:
sql => SQL
ip => IP
(try to make the doc formal and professional)

Reply:
ok